### ISSUE TITLE
Printing (to Kitchen) is not automatic for ADDITIONAL ORDERS

### ISSUE THREAD

**Issue by TeaConcepts** [nexopos/v3-issues#252]
**Username** : MOJOTOJO

**Describe the bug**
Printing (to Kitchen) is not automatic for ADDITIONAL ORDERS. Works fine on the original/first order.

**To Reproduce**
We're Using NORMAL printer only......USB Thermal Printer

1. Create an order for Table 1
2. Send to Kitchen
--> POS automatically launches print dialogue (this is expected)
3. Open TABLE 1 again, add order, then send to kitchen again

**Expected behavior**
What's Expected: Printer (or dialogue) fires up automatically and prints just the additional orders of Table 1.
Actual Behavior: Printer (or dialogue) doesn't fire up automatically, instead I am presented with the Tables list screen (which doesn't make sense). I can't seem to print just the additional orders.

**Screenshots**
Video Demo: https://1drv.ms/u/s!AjHDfujP9TT_gftg17JWQgDLi06_MA?e=xRWd6Y

**Environment (please complete the following information):**
Versions Installed: https://i.imgur.com/io4W7x9.jpg

 - PHP version: 7.2.19
 - MySQL version: 5.0.12
 - Devise used: Desktop + Chrome and Firefox

**Comment by Blair2004** [comment]
Hi, thank you for updating us here and nice first issue.
Question : are you using Nexo Print Server for kitchen printing ?

**Comment by TeaConcepts** [comment]
> 
> 
> Hi, thank you for updating us here and nice first issue.
> Question : are you using Nexo Print Server for kitchen printing ?

thanks. and newp, just vanila NORMAL printing mode off the browser and a USB connected thermal printer.

**Comment by Blair2004** [comment]
By default, Google Cloud Print or Nexo Print Server (only) are used to send the print to the kitchen. If you're however talking about the sale receipt (because both are different), we'll investigate. But, we would like to be sure you're talking about the kitchen receipt or the sale receipt?

**Comment by TeaConcepts** [comment]
I'm confused about the limitation you described. 
I have clearly show on video demo what my issue is..
it's MAKING ADDITIONAL ORDERS and SENDING THOSE ORDERS TO KITCHEN 

This again is clearly demonstrated on the video demo, i dont know how else to explain...

I dont , and dont want to, use either either GCP or Nexpo Print Server ... 

again, 
When making the order for the first time , pressing "TO KITCHEN" opens the printer dialogue just fine and we're able to print...

ADDING ORDER and pressing "TO KITCHEN" again does not work as it no longer brings up the printer dialogue.

**Comment by Blair2004** [comment]
Hi, 
the review on your issue is based on the information you provides on your description. You have been mentioning "Kitchen" printing which involves receiving a print job at the kitchen (completely different) from sale receipt (or POS printing). If both are made almost at the same time, both aren't similar. The POS printing includes prices and kitchen printing is sent to the right kitchen (according to your categories mapping or not) with only products and modifiers.

So, your issue (if well understood) point out that a sales receipt should be printed after a new items is send to the kitchen. Because, kitchen receipt only works with either Nexo Print Server or Google Cloud Print by default. In such case, we have intentionally disabled the sale receipt printing for new items that cause a waste of paper especially since a direct (what you're doing at the end of your video) could be take to make the sale receipt to print.

To summarize, what you're using so far is not the kitchen printing, but the pos printing and what you're facing is completely intentional. We advice you to either use Google Cloud Print (free) or Nexo Print Server (premium + more features) to have the kitchen printing.

For more informations about how it works : https://nexopos.com/how-does-the-split-print-works-on-gastro/

![screenshot-github com-2019 10 08-03-56-29](https://user-images.githubusercontent.com/5265663/66361843-ab44a880-e980-11e9-95b3-4f434ff8bc56.png)

**Comment by TeaConcepts** [comment]
Thank you for explaining and now I understand that this is an intended behavior. 
But if I may poke further, we are using the same USB printer for counter and for kitchen as well... basically the process is as follows.

1. Order is taken, then order is sent to kitchen -> all we need is 1 copy of the receipt to give to be given to kitchen by hand 

2. Then for additional orders, we again NEED TO PRINT "additional orders" only on the same USB printer.. 

3. Then finally after payment, cashier prints out a receipt... 


Both 1 and 3 works fine. Given our setup, how do you suggest we achieve #2 ? 

Also something confusing still... if KITCHEN PRINTER is enabled thru for GCP or Nexo Print Server only, and we dont have both, why is it that our current setup enables us to print (brings up the PRINT DIALOGUE) when pressing "TO KITCHEN" option at first order taking? If the above statement is true then printing should only be offer to user upon checkout/payment of the bill.

**Comment by TeaConcepts** [comment]
Alright thank you and i hope to see this feature added really soon. I'm also a full stack dev, been one for over 20 yrs. i can actually hack code into your source to EASILY achieve this but i dont wanna touch your source nor do i wanna have to deal w/ maintaining code during updates and what nots. So i hope you guys can spare a a few moments of dev time to squeeze this feature in.  thank you

**Comment by timothyte** [comment]
Hi, any news on this ? We're using your POS now and every time a customer orders additional things like drinks and sides, the whole list still prints out, it's a huge waste of paper, and hoping to get an update for this :) Cheers!

**Comment by Blair2004** [comment]
Please consider opening another issue @timothyte so that we can proceed with you.

**Comment by timothyte** [comment]
> Please consider opening another issue @timothyte so that we can proceed with you.

My issue is exactly the same as thread starter. need a way to print ONLY additional orders, not entire list.

**Comment by TeaConcepts** [comment]
@Blair2004  - man it's been a while. we still dont have a fix for this??

**Comment by TeaConcepts** [comment]
helloooooo?? we still can't use the POS properly due to this bug.. where's the support here??? :(

**Comment by Blair2004** [comment]
Hi, very sorry for the delay. In fact the feature has been scheduled for NexoPOS 4.x. However, since that make it impossible for you to use the system, I'll consider adding that on Gastro.

I'll keep you updated on the progress.

Thank you

**Comment by TeaConcepts** [comment]
thanks blair2004 - looking forward
